libpod/sqlite_state.go

```diff
@@ -0,0 +1,2535 @@
+	// if err != nil {
+	// 	return fmt.Errorf("marshalling new configuration JSON for pod %s: %w", pod.ID(), err)
+	// }
+
+	// db, err := s.getDBCon()
+	// if err != nil {
+	// 	return err
+	// }
+	// defer s.deferredCloseDBCon(db)
+
+	// err = db.Update(func(tx *bolt.Tx) error {
+	// 	podBkt, err := getPodBucket(tx)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+
+	// 	podDB := podBkt.Bucket([]byte(pod.ID()))
+	// 	if podDB == nil {
+	// 		pod.valid = false
+	// 		return fmt.Errorf("no pod with ID %s found in DB: %w", pod.ID(), define.ErrNoSuchPod)
+	// 	}
+
+	// 	if err := podDB.Put(configKey, newCfgJSON); err != nil {
+	// 		return fmt.Errorf("updating pod %s config JSON: %w", pod.ID(), err)
+	// 	}
+
+	// 	return nil
+	// })
+	// return err
+}
+
+// RewriteVolumeConfig rewrites a volume's configuration.
+// WARNING: This function is DANGEROUS. Do not use without reading the full
+// comment on this function in state.go.
+// TODO TODO TODO
+func (s *SQLiteState) RewriteVolumeConfig(volume *Volume, newCfg *VolumeConfig) error {
+	if !s.valid {
+		return define.ErrDBClosed
+	}
+
+	if !volume.valid {
+		return define.ErrVolumeRemoved
+	}
+
+	return define.ErrNotImplemented
+
+	// newCfgJSON, err := json.Marshal(newCfg)
+	// if err != nil {
+	// 	return fmt.Errorf("marshalling new configuration JSON for volume %q: %w", volume.Name(), err)
+	// }
+
+	// db, err := s.getDBCon()
+	// if err != nil {
+	// 	return err
+	// }
+	// defer s.deferredCloseDBCon(db)
+
+	// err = db.Update(func(tx *bolt.Tx) error {
+	// 	volBkt, err := getVolBucket(tx)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+
+	// 	volDB := volBkt.Bucket([]byte(volume.Name()))
+	// 	if volDB == nil {
+	// 		volume.valid = false
+	// 		return fmt.Errorf("no volume with name %q found in DB: %w", volume.Name(), define.ErrNoSuchVolume)
+	// 	}
+
+	// 	if err := volDB.Put(configKey, newCfgJSON); err != nil {
+	// 		return fmt.Errorf("updating volume %q config JSON: %w", volume.Name(), err)
+	// 	}
+
+	// 	return nil
+	// })
+	// return err
+}
+
+// Pod retrieves a pod given its full ID
+// TODO TODO TODO
+func (s *SQLiteState) Pod(id string) (*Pod, error) {
+	if id == "" {
+		return nil, define.ErrEmptyID
+	}
+
+	if !s.valid {
+		return nil, define.ErrDBClosed
+	}
+
+	return nil, define.ErrNotImplemented
+
+	// podID := []byte(id)
+
+	// pod := new(Pod)
+	// pod.config = new(PodConfig)
+	// pod.state = new(podState)
+
+	// db, err := s.getDBCon()
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// defer s.deferredCloseDBCon(db)
+
+	// err = db.View(func(tx *bolt.Tx) error {
+	// 	podBkt, err := getPodBucket(tx)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+
+	// 	return s.getPodFromDB(podID, pod, podBkt)
+	// })
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// return pod, nil
+}
+
+// LookupPod retrieves a pod from a full or unique partial ID, or a name.
+func (s *SQLiteState) LookupPod(idOrName string) (*Pod, error) {
+	if idOrName == "" {
```

Comment:
this same boilerplate is used each time. i could some beauty in something like:
```
if s.validate(idOrName); err != nil { return nil,  err}
````